File: cpp/1189.cpp
#include <iostream>
#include <stdio.h>

using namespace std;

float sum(float m[], int s, int e)
{
    float sum = 0;

    for (s; s < e; s++) {
        sum += m[s];
    }

    return sum;
}

int main()
{
    int index = 12, step = 1, swi = 2, i;
    float m[144], result = 0;
    char op;
    cin >> op;

    for (i = 0; i < 144; i++) {
        cin >> m[i];
    }

    while (step > -1) {
        result += sum(m, index, index + step);
        index += 12;

        if (swi == 2) {
            step++;
        } else if (swi == 0) {
            step--;
        }

        if (step == 5) {
            swi--;
        }
    }

    if (op == 'S') {
        printf("%.1f\n", result);
    } else {
        printf("%.1f\n", result / 30);
    }

    return 0;
}

File: py/1189.py
operation = input()
matrix = []

i = 0
while i < 144:
    matrix.append(float(input()))
    i += 1

result = 0
index = 12
step = 1
switch = 2

while True:
    result += sum(matrix[index : index + step : 1])
    
    index += 12

    if switch == 2:
        step += 1
        
    elif switch == 0:
        step -= 1

    if step == 5:
        switch -= 1

    if step == -1:
        break

if operation == 'S':
    print(f'{(result):.1f}')

else:
    print(f'{(result / 30):.1f}')